### PR TITLE
Refactored the Pipelines navigation.

### PR DESCRIPTION
**Pull request by sarahmaddox**
We've created quite a few Pipelines docs recently. It's time to refactor the left-hand navigation panel for Pipelines. This change affects the index pages for each section too.

Fixes https://github.com/kubeflow/website/issues/818

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/820)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
Preview - see all the docs in this section:
https://deploy-preview-820--competent-brattain-de2d6d.netlify.com/docs/pipelines/

Examples of redirects:

* https://deploy-preview-820--competent-brattain-de2d6d.netlify.com/docs/pipelines/pipelines-overview/
* https://deploy-preview-820--competent-brattain-de2d6d.netlify.com/docs/pipelines/concepts/
* https://deploy-preview-820--competent-brattain-de2d6d.netlify.com/docs/pipelines/concepts/pipeline/
* https://deploy-preview-820--competent-brattain-de2d6d.netlify.com/docs/pipelines/preemptible/

**Comment by sarahmaddox** [comment]
This PR is now ready for review. Adding a hold to allow for multiple reviewers

/hold
/assign @IronPan

**Comment by IronPan** [comment]
I really like the way this doc is structured. Thank you Sarah
/lgtm
/approve

File: content/docs/pipelines/sdk/gcp/_index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
++++
+title = "GPC-specific Uses of the SDK"
```
Comment (alecglassford) [inline content/docs/pipelines/sdk/gcp/_index.md]: Typo: GPC --> GCP

Comment (sarahmaddox) [reply content/docs/pipelines/sdk/gcp/_index.md]: Ooh thanks!

**Pull request merged**